### PR TITLE
Remove deprecated HEARTBEAT table

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
The failing test is originally flaky

File: src/ray/protobuf/gcs.proto [hunk 1]
```diff
@@ -56,7 +55,6 @@ enum TablePubsub {
   NODE_PUBSUB = 4;
   OBJECT_PUBSUB = 5;
   ACTOR_PUBSUB = 6;
-  HEARTBEAT_PUBSUB = 7;
   RESOURCE_USAGE_BATCH_PUBSUB = 8;
```
Comment (rkooo567) [inline src/ray/protobuf/gcs.proto:58]: Can you update the integer?

**Review by rkooo567 — CHANGES_REQUESTED** [review]

File: src/ray/protobuf/gcs.proto (hunk 1 above)
Comment (rkooo567) [reply src/ray/protobuf/gcs.proto:58]: e.g., `RESOURCE_USAGE_BATCH_PUBSUB =7`

Comment (WangTaoTheTonic) [reply src/ray/protobuf/gcs.proto:58]: No problem. The reason why I didn't update them is that I saw the number below also has a gap.
I'll update all of them.

**Review by rkooo567 — APPROVED** [review]

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**